cache/cache_entry_roles.h

```diff
@@ -14,6 +14,8 @@
 namespace ROCKSDB_NAMESPACE {
 
 // Classifications of block cache entries, for reporting statistics
+// Adding new enum to this class requires corresponding updates to
+// kCacheEntryRoleToCamelString and kCacheEntryRoleToHyphenString
```

Comment:
[Note]
I tried also adding a `static_assert(kCacheEntryRoleToCamelString.size(), kNumCacheEntryRoles)` in   cache/cache_entry_roles.c but the assertion doesn't work and I haven't figured out the reason why ...

Reply:
std::array has size specified in template parameter, and you're allowed to initialize fewer elements. I wanted to add a similar static assertion, but C++ doesn't make it easy.

Reply:
> std::array has size specified in template parameter, and you're allowed to initialize fewer elements.

That's exactly why my static_assert does not work! The only thing I can think of right now is to do some stripped string match between the "kMisc" and "kNumCacheEntryRoles-1"-th element of kCacheEntryRoleToCamelString but accessing the uninitialized element of an array (i.e, "kNumCacheEntryRoles-1"-th element of kCacheEntryRoleToCamelString if we forgot to update it) can give undefined behavior.

So now I can see more why previous attempt to add such a static assertion failed and hope that future developers find my comment useful!

Reply:
Fortunately, UBSAN (and friends?) are pretty good at picking up on using uninitialized memory.